### DISCUSSION TITLE
Planned Brown-out and Steps to Discourage Continued Usage of Bionic Builders

### DISCUSSION THREAD

**Discussion by dmikusa** [paketo-buildpacks/feedback#224] [Maintainer Votes]
## Problem Statement

On May 31st 2023, Ubuntu 18.04 (Bionic Beaver) went out of support for OSS customers ([source](https://ubuntu.com/blog/18-04-end-of-standard-support)). At the same time, the Paketo Project [stopped supporting its Bionic Stacks and Builders based on Ubuntu 18.04](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0057-bionic-eos.md). This was further communicated in a [blog post on July 28, 2023](https://blog.paketo.io/posts/bionic-eos/).

As of today, we are still seeing a number of users performing builds against the Bionic stacks. This comes up in forum posts, Stack Overflow questions, Slack questions, and other channels. A primary driver is that [Spring Boot Users will continue to default to the Bionic Stack until Spring Boot 3.2](https://spring.io/blog/2023/09/22/paketo-buildpacks-bionic-end-of-support). This will mean at a minimum, a number of users will continue defaulting on Bionic until at least the end of November, but potentially longer as not every user will immediately upgrade when Spring Boot 3.2 is available. Users that stay on older versions of Spring Boot will continue to default to the Bionic builder.

## Proposed Plan

1. On Wed Nov 15th, we will post a blog post advertising an upcoming brownout and tag change. We will use this to alert users of the upcoming change, explain the actions that they need to take to avoid having a problem during the brownout and use it as a point of reference for anyone who might experience issues during the brownout.

2. We will schedule a brownout for Monday, Nov 20th, from 10 am EST to 2 pm EST. We believe this will cover a time range across EMEA, US East, and US West. We hope this will be long enough time period that users will notice build failures and be encouraged to upgrade.

3. This will be implemented by temporarily removing the image tags from the Bionic builder images (full, base, and tiny). Once the brownout is complete, the image tags will be restored. This will restore user builds to a working order.

4. Starting Jan 1, 2024 we will permanently remove the image tags from the Bionic builder images (full, base, and tiny). By this point in time, we hope users have migrated off of it to Jammy. If for some reason a user must stay on the Bionic builder, we will add a tag back called `unsafe`. Users may switch to the `unsafe` tag, and opt to continue to use the Bionic builder.

## Required Action

To move forward with this plan, steering committee members and all project team maintainers need to vote. Standard voting rules apply, and the plan needs to be approved. If the resolution is not approved, no action will be taken.

The vote is open until the end of the day on Tue Nov 14th, 2023.

Please do not vote unless you are a steering committee member or project maintainer. Other votes will not count but will complicate the voting process.

**Comment by dmikusa** [comment]
@paketo-buildpacks/steering-committee
@paketo-buildpacks/builders-maintainers 
@paketo-buildpacks/go-maintainers 
@paketo-buildpacks/php-maintainers 
@paketo-buildpacks/java-maintainers 
@paketo-buildpacks/ruby-maintainers 
@paketo-buildpacks/rust-maintainers 
@paketo-buildpacks/python-maintainers 
@paketo-buildpacks/app-monitoring-maintainers 
@paketo-buildpacks/stacks-maintainers 
@paketo-buildpacks/nodejs-maintainers 
@paketo-buildpacks/dotnet-core-maintainers 
@paketo-buildpacks/content-maintainers 
@paketo-buildpacks/tooling-maintainers 
@paketo-buildpacks/web-servers-maintainers 
@paketo-buildpacks/utilities-maintainers 

I think that's all, sorry if I missed any.

**Reply by dmikusa** [reply]
Please also add a Rocket Ship emoji to the post when you've voted. GitHub does not allow us to see your specific vote, but by adding a reaction we can make sure that only the appropriate folks are voting. Thanks

**Reply by joshuatcasey** [reply]
I approved, but I do wonder if 4 hours is enough to ensure the desired effect. Perhaps we could also add a deprecation notice to https://hub.docker.com/r/paketobuildpacks/builder?

**Reply by dmikusa** [reply]
Yes, that's an excellent idea. Thanks @joshuatcasey

**Reply by joshuatcasey** [reply]
Cool. I suspect that's where I would go if the tag I wanted was no longer available.

**Comment by dmikusa** [comment]
Thanks everyone for taking the time to vote. With 9 affirmative votes, we have surpassed the required 2/3rd votes. We will proceed with the plan outlined starting tomorrow with the blog post.
